caddy/restartinproc.go

```diff
@@ -23,6 +26,7 @@ func restartInProc(newCaddyfile Input) error {
 		} else {
 			wg.Done() // take down our barrier
 		}
+		go notify.IsReloading(false).Tell()
```

Comment:
You probably need one more `go notify.IsReloading(false).Tell()` before `return nil`.
Or just defer it at function start.